src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol_ImplementationChecks.cs

```diff
@@ -985,11 +985,10 @@ private void CheckNewModifier(Symbol symbol, bool isNew, DiagnosticBag diagnosti
 
                             for (int i = 0; i < overridingParameters.Length; i++)
                             {
-                                var overridenParameterType = overriddenParameters[i].TypeWithAnnotations;
+                                var overriddenParameterType = overriddenParameters[i].TypeWithAnnotations;
                                 var overridingParameterType = overridingParameters[i].TypeWithAnnotations;
-                                if (!overridingParameterType.Equals(overridenParameterType,
-                                                                         TypeCompareKind.AllIgnoreOptions & ~(TypeCompareKind.IgnoreNullableModifiersForReferenceTypes)) &&
-                                    overridingParameterType.Equals(overridenParameterType, TypeCompareKind.AllIgnoreOptions))
+                                var conversions = compilation.Conversions.WithNullability(true);
```

Comment:
>compilation.Conversions.WithNullability(true) [](start = 50, length = 45)

I'm thinking we should cache this object on the compilation.

Reply:
The instance is cached in `compilation.Conversions._lazyOtherNullability`.